windows-driver-docs-pr/bluetooth/microsoft-defined-bluetooth-hci-commands-and-events.md

```diff
@@ -267,23 +269,76 @@ HCI_VS_MSFT_LE_Monitor_Advertisement requests that the controller starts monitor
+
 | Command | Code | Command parameters | Return parameters |
 |--|--|--|--|
-| HCI_VS_MSFT_LE_Monitor_Advertisement | Chosen base code | Subcommand_opcode,<br>RSSI_threshold_high,<br>RSSI_threshold_low,<br>RSSI_threshold_low_time_interval,<br>RSSI_sampling_period,<br>Condition_type,<br>\<Condition Parameters\> | Status,<br>Subcommand_opcodee,<br>Monitor_Handle |
+| HCI_VS_MSFT_LE_Monitor_Advertisement [v2] | Chosen base code | Subcommand_opcode_v2,<br>RSSI_threshold_high,<br>RSSI_threshold_low,<br>RSSI_threshold_low_time_interval,<br>RSSI_sampling_period,<br>Monitor_options,<br>Advertisement_report_filtering_options,<br>Peer_device_address,<br>Peer_device_address_type,<br>Peer_device_IRK,<br>Condition_type,<br>\<Condition Parameters\> | Status,<br>Subcommand_opcode,<br>Monitor_Handle |
+| HCI_VS_MSFT_LE_Monitor_Advertisement [v1] | Chosen base code | Subcommand_opcode_v1,<br>RSSI_threshold_high,<br>RSSI_threshold_low,<br>RSSI_threshold_low_time_interval,<br>RSSI_sampling_period,<br>Condition_type,<br>\<Condition Parameters\> | Status,<br>Subcommand_opcode,<br>Monitor_Handle |
 
 The controller shall generate a Command Complete event in response to this command. The status value should be set to zero if the controller can begin monitoring, or a nonzero status otherwise.
 If the controller doesn't support RSSI monitoring for LE Advertisements, it shall ignore the *RSSI_threshold_high*, *RSSI_threshold_low*, *RSSI_threshold_low_time_interval*, and *RSSI_sampling_period* parameter values.
 
 This state diagram shows the transition states on the controller when monitoring RSSI for an advertisement.
 
-:::image type="content" source="images/HCI_VS_MSFT_LE_Monitor_Advertisement_State_Diagram.png" alt-text="State diagram for HCI_VS_MSFT_LE_Monitor_Advertisement.":::
+:::image type="content" source="images/HCI_VS_MSFT_LE_Monitor_Advertisement_State_Diagram.svg" alt-text="State diagram for HCI_VS_MSFT_LE_Monitor_Advertisement.":::
+
+The controller shall begin monitoring an advertisement only when the received RSSI is greater than or equal to *RSSI_threshold_high* for a particular device and the *Monitor_options* match (see below).  The controller shall generate an [HCI_VS_MSFT_LE_Monitor_Device_Event][ref_HCI_VS_MSFT_LE_Monitor_Device_Event] with *Monitor_state* set to 1 and *Monitor_handle* set to the handle for this *Condition*, to notify the host that the controller is monitoring this particular device for *Condition*. Additionally, the Controller shall propagate the first advertisement report of a monitored advertisement to the Host only when the *Advertisement_report_filter_options* match (see below).
+
+The *Monitor_options* for a filter are considered a match based on the following logic:
+
+```
```

Comment:
Be sure to mark all the code blocks with the dev language so the syntax coloring is rendered properly. A C++code block looks like this:

\```cpp
 // C++ code goes here
\```

Reply:
Actually, this looks like pseudo-code. We don't really have a way to specify pseudo-code.

Reply:
Thanks Mark! These are just pseudocode to help explain the logic. I wonder if there's a markdown code block language that would give them some basic highlighting...? I could try a few to see what happens.

Reply:
Unfortunately, `cpp` behaves pretty strangely with the sorta-English fragments in there. I should definitely indicate that it isn't meant to be "real" code, but I'm at a loss for a lexer that nicely highlights the operators without tripping over everything else.

Reply:
I guess we just need to leave the dev language unspecified. The only other solution I can think is to not use a code block at all, just describe the logic in the text.

Reply:
Yeah, I originally tried explaining the logic in text, but it got pretty hard to express all of the conditions..